eclass/autotools.eclass

```diff
@@ -312,7 +312,13 @@ eautoreconf() {
 	else
 		eautoconf --force
 	fi
-	[[ ${AT_NOEAUTOHEADER} != "yes" ]] && eautoheader
+	if [[ ${AT_NOEAUTOHEADER} != "yes" ]] ; then
+		if [[ ${WANT_AUTOCONF} == "2.1" ]] ; then
```

Comment:
But what if lesser than 2.1?

Reply:
It's not valid to set < 2.1: https://github.com/gentoo/gentoo/blob/ab3027da217a4e32a853a41ae37c013457556f0e/eclass/autotools.eclass#L124-L153

This condition is copied from eautoconf:
https://github.com/gentoo/gentoo/blob/ab3027da217a4e32a853a41ae37c013457556f0e/eclass/autotools.eclass#L310-L314